src/data_readers/data_reader_jag_conduit.cpp

```diff
@@ -943,98 +762,126 @@ void data_reader_jag_conduit::load() {
+      }
     }
-    const size_t p = static_cast<size_t>(m_my_partition);
-    const size_t idx_start = min_num_files_per_partition * p
-                           + ((p >= one_more)? one_more : p);
 
-    const size_t idx_end = idx_start + min_num_files_per_partition
-                           + ((p < one_more)? 1u : 0u);
-    std::vector<std::string> filenames_partitioned(filenames.begin()+idx_start, filenames.begin()+idx_end);
-    filenames = filenames_partitioned;
-  }
-  const size_t num_files_to_load =
-    (m_max_files_to_load > 0u)? std::min(m_max_files_to_load, filenames.size()) : filenames.size();
+    if (!m_is_data_loaded) {
+      m_is_data_loaded = true;
 
-  filenames.resize(num_files_to_load);
+      if (m_scalar_keys.size() == 0u) {
+        set_all_scalar_choices(); // use all by default if none is specified
+      }
+      check_scalar_keys();
 
-  double tm1 = get_time();
+      if (m_input_keys.size() == 0u) {
+        set_all_input_choices(); // use all by default if none is specified
+      }
+      check_input_keys();
 
-  // Reserve m_valid_samples
-  const size_t max_num_files_to_load_per_rank = (num_files_to_load + num_readers - 1u) / num_readers;
-  bool valid_samples_reserved = false;
-  size_t idx = static_cast<size_t>(0ul);
-
-  for (size_t n = my_rank; (n < num_files_to_load) && (my_rank < num_readers); n += num_readers) {
-    load_conduit(filenames[n], idx);
-    if (!valid_samples_reserved) {
-      // reserve the sufficient capacity estimated assuming that files have the same number of samples
-      m_valid_samples.reserve(m_valid_samples.size() * (max_num_files_to_load_per_rank + 1u));
-      valid_samples_reserved = true;
-    }
-    if (is_master()) {
-      std::cerr << "time to load: " << n + num_readers << " files: " << get_time() - tm1 << std::endl;
+      check_image_data();
     }
+    m_shuffled_indices.resize(m_sample_list.size());
   }
+
+  std::iota(m_shuffled_indices.begin(), m_shuffled_indices.end(), 0);
+
+  select_subset_of_data();
+}
+
+void data_reader_jag_conduit::load_list_of_samples(const std::string sample_list_file) {
+  // load the sample list
+  double tm1 = get_time();
+  m_sample_list.load(sample_list_file);
+  double tm2 = get_time();
+
   if (is_master()) {
-    std::cerr << "time to load conduit files: " << get_time() - tm1
-              << "  number of valid local samples at the master rank: " << m_valid_samples.size()
-              << " local reader id=" << get_local_id(get_role()) << " for " << get_role()
-              << " leading reader=" << m_leading_reader << std::endl;
+    std::cout << "Time to load sample list: " << tm2 - tm1 << std::endl;
   }
+}
+
+void data_reader_jag_conduit::load_list_of_samples_from_archive(const std::string& sample_list_archive) {
+  // load the sample list
+  double tm1 = get_time();
+  std::stringstream ss(sample_list_archive); // any stream can be used
+
+  cereal::BinaryInputArchive iarchive(ss); // Create an input archive
```

Comment:
I have never heard this serialization library before, but it seems very useful and easy to use. I had to quickly glance at their web documentation to get a idea. One question I had was that is it ok to use binary archive with stringstream? They mention to set std::ios::binary when using with stream for binary archive.

Reply:
I don't see how to do that with a stringstream.  Also pulling from their example I didn't see that trait set on a stringstream.